lib/s3routes/routes/routeGET.js

```diff
@@ -125,6 +125,13 @@ function routerGET(request, response, api, log, statsClient, dataRetrievalFn) {
                     return routesUtils.responseXMLBody(err, xml, response, log,
                         corsHeaders);
                 });
+        } else if (request.query.retention !== undefined) {
```

Comment:
This code block is virtually identical to the others minus like one string, it should be possible to map these to an object instead of if blocks per request. Later of course, just thought I would mention it.